plugin/plaintextmodel.cpp

```diff
@@ -264,6 +277,65 @@ bool PlainTextModel::readLines(
     return false;
 }
 
+QTextCodec *PlainTextModel::detectCodec(const char *encodedCharacters, qint64 length)
+{
+    QLocale locale;
+    QTextCodec *codec = QTextCodec::codecForName("UTF-8");
+
+    UErrorCode error = U_ZERO_ERROR;
+    if (UCharsetDetector * const detector = ucsdet_open(&error)) {
+        ucsdet_setText(detector, encodedCharacters, length, &error);
+
+        int32_t count = 0;
+        if (U_FAILURE(error)) {
+            qWarning() << "Unable to detect text encoding" << u_errorName(error);
+        } else if (const UCharsetMatch ** const matches = ucsdet_detectAll(detector, &count, &error)) {
+            int bestConfidenceICU = 0;
+            QTextCodec * bestMatchICU = nullptr;
+
+            for (int32_t i = 0; i < count; ++i) {
+                if (QTextCodec *detectedCodec = QTextCodec::codecForName(ucsdet_getName(matches[i], &error))) {
+                    bestConfidenceICU = ucsdet_getConfidence(matches[i], &error);
+                    bestMatchICU = detectedCodec;
+
+                    break;
+                }
+            }
+
+            if(bestMatchICU) {
+                QString localeName = locale.name();
+
+                int bestConfidenceLan = 0;
+                QTextCodec * bestMatchLan = nullptr;
```

Comment:
I'd usually avoid abbreviations.

Reply:
What do you mean? There are no acronyms in the name

Reply:
"lan" is not a word, or well could be used for local area network. -> bestMatchLanguage or something like that.

Reply:
Done

Reply:
`lang` could be an appropriate abbreviation for `language` but `lan` is definitely too short, I second that.

Reply:
Yes. I've already corrected it